GoogleDataTransport/GDTCORLibrary/Public/GDTCORStorageProtocol.h

```diff
@@ -36,6 +38,31 @@ NS_ASSUME_NONNULL_BEGIN
 /** Removes the events from storage. */
 - (void)removeEvents:(NSSet<NSNumber *> *)eventIDs;
 
+/** Persists the given data with the given key.
+ *
+ * @param data The data to store.
+ * @param key The unique key to store it to.
+ * @param onComplete An block to be run when storage of the data is complete.
+ */
+- (void)storeLibraryData:(NSData *)data
```

Comment:
I wonder what is the intended usage of the storage. Is the client still expected to use `NSKeyedArchiver` to create data from e.g. an `NSArray` instance? If so, do you think we may use `id<NSSecureCoding>` instead of `NSData` type for values and encapsulate `NSKeyed(Un)Archiver` logic in the storage?

Reply:
It's meant to store library state. A caller _can_ implement NSSecureCoding to generate its bytes, but it's not a hard requirement. How the caller generates bytes is a decision left to that specific class. Because several of the callers will be singletons, it's created a lot of NSCoding weirdness that this can help to avoid. I prefer to keep it as NSData because then NSCoding can be used when wanted, or, it can just be a string or whatever.